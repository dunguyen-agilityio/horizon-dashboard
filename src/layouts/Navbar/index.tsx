'use client';

// Libs
import { usePathname } from 'next/navigation';

// Components
import {
  Navbar as NavComponent,
  NavbarContent,
  NavbarBrand,
  NavbarItem,
} from '@nextui-org/navbar';
import { Banner, Brand, NavItem } from '@/components';
import { Divider } from '@nextui-org/divider';

// Constants
import { NAV_ROUTES, AUTH_ROUTES } from '@/constants/routes';

interface NavbarProps {
  isAuthenticated: boolean;
}

const Navbar = ({ isAuthenticated }: NavbarProps) => {
  const pathname = usePathname();

  return (
    <NavComponent
      className="bg-white w-navbar dark:bg-indigo h-screen"
      classNames={{
        wrapper: 'px-0 flex-col h-full',
      }}
    >
      <NavbarBrand className="flex-grow-0">
        <div className="flex flex-col w-[290px]">
          <Brand />
          <Divider className="bg-zinc-100 dark:bg-slate-700" />
        </div>
      </NavbarBrand>
      <NavbarContent className="flex flex-1 w-full flex-col items-center justify-between pt-[38px] gap-5">
        {NAV_ROUTES.map(({ href, icon, title }, index) => {
          if (isAuthenticated && href === AUTH_ROUTES.SIGN_IN) return null;
          return (
            <NavbarItem key={`${title}-${index}`} className="w-full">
              <NavItem
                href={href}
                icon={icon}
                label={title}
                isActive={href === pathname}
              />
            </NavbarItem>
          );
        })}
        <li className="flex-1 flex items-end mb-6">
          <Banner />
        </li>
      </NavbarContent>
    </NavComponent>
  );
};

export default Navbar;
