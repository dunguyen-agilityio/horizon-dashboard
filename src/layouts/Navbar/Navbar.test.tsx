import { render } from '@testing-library/react';
import Navbar from '.';

jest.mock('next/navigation', () => ({
  usePathname: jest.fn().mockImplementation(() => '/'),
}));

describe('Navbar tests', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Should match snapshot', () => {
    const { container } = render(<Navbar />);
    expect(container).toMatchSnapshot();
  });
});
