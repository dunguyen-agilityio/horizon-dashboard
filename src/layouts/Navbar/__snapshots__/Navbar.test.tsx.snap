// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar tests Should match snapshot 1`] = `
<div>
  <nav
    class="bg-white flex flex-col justify-between dark:bg-indigo w-[240px] lg:w-[290px] h-[1152px]"
  >
    <div
      class="flex flex-col pt-[38px] gap-5"
    >
      <a
        class="flex justify-between"
        href="/dashboard"
      >
        <div
          class="flex w-full items-center gap-3 pl-8"
        >
          <i
            class="w-6 h-6 flex justify-center items-center dark:fill-white fill-secondary"
            data-testid="nav-icon"
          >
            <svg
              height="17"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 16v-5h4v5c0 .55.45 1 1 1h3c.55 0 1-.45 1-1V9h1.7c.46 0 .68-.57.33-.87L10.67.6c-.38-.34-.96-.34-1.34 0L.97 8.13c-.34.3-.13.87.33.87H3v7c0 .55.45 1 1 1h3c.55 0 1-.45 1-1Z"
              />
            </svg>
          </i>
          <p
            class="text-primary dark:text-white text-medium leading-xl font-medium text-secondary"
          >
            Dashboard
          </p>
        </div>
      </a>
      <a
        class="flex justify-between"
        href="/nft-marketplace"
      >
        <div
          class="flex w-full items-center gap-3 pl-8"
        >
          <i
            class="w-6 h-6 flex justify-center items-center dark:fill-white fill-secondary"
            data-testid="nav-icon"
          >
            <svg
              height="20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.55 11c.75 0 1.41-.41 1.75-1.03l3.58-6.49A.996.996 0 0 0 19.01 2H4.21l-.94-2H0v2h2l3.6 7.59-1.35 2.44C3.52 13.37 4.48 15 6 15h12v-2H6l1.1-2h7.45ZM5.16 4h12.15l-2.76 5H7.53L5.16 4ZM6 16c-1.1 0-1.99.9-1.99 2S4.9 20 6 20s2-.9 2-2-.9-2-2-2Zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2Z"
              />
            </svg>
          </i>
          <p
            class="text-primary dark:text-white text-medium leading-xl font-medium text-secondary"
          >
            NFT Marketplace
          </p>
        </div>
      </a>
      <a
        class="flex justify-between"
        href="/contributor"
      >
        <div
          class="flex w-full items-center gap-3 pl-8"
        >
          <i
            class="w-6 h-6 flex justify-center items-center dark:fill-white fill-secondary"
            data-testid="nav-icon"
          >
            <svg
              height="18"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M2.467 5.733H2.7c.898 0 1.633.735 1.633 1.634v8.166c0 .899-.735 1.634-1.633 1.634h-.233a1.638 1.638 0 0 1-1.634-1.634V7.367c0-.899.735-1.634 1.634-1.634ZM9 .833c.898 0 1.633.735 1.633 1.634v13.066c0 .899-.735 1.634-1.633 1.634a1.638 1.638 0 0 1-1.633-1.634V2.467c0-.899.735-1.634 1.633-1.634Zm6.533 9.334c.899 0 1.634.735 1.634 1.633v3.733c0 .899-.735 1.634-1.634 1.634a1.638 1.638 0 0 1-1.633-1.634V11.8c0-.898.735-1.633 1.633-1.633Z"
              />
            </svg>
          </i>
          <p
            class="text-primary dark:text-white text-medium leading-xl font-medium text-secondary"
          >
            Contributor
          </p>
        </div>
      </a>
      <a
        class="flex justify-between"
        href="/kanban"
      >
        <div
          class="flex w-full items-center gap-3 pl-8"
        >
          <i
            class="w-6 h-6 flex justify-center items-center dark:fill-white fill-secondary"
            data-testid="nav-icon"
          >
            <svg
              height="18"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 10h6c.55 0 1-.45 1-1V1c0-.55-.45-1-1-1H1C.45 0 0 .45 0 1v8c0 .55.45 1 1 1Zm0 8h6c.55 0 1-.45 1-1v-4c0-.55-.45-1-1-1H1c-.55 0-1 .45-1 1v4c0 .55.45 1 1 1Zm10 0h6c.55 0 1-.45 1-1V9c0-.55-.45-1-1-1h-6c-.55 0-1 .45-1 1v8c0 .55.45 1 1 1ZM10 1v4c0 .55.45 1 1 1h6c.55 0 1-.45 1-1V1c0-.55-.45-1-1-1h-6c-.55 0-1 .45-1 1Z"
              />
            </svg>
          </i>
          <p
            class="text-primary dark:text-white text-medium leading-xl font-medium text-secondary"
          >
            Kanban
          </p>
        </div>
      </a>
      <a
        class="flex justify-between"
        href="/profile"
      >
        <div
          class="flex w-full items-center gap-3 pl-8"
        >
          <i
            class="w-6 h-6 flex justify-center items-center dark:fill-white fill-secondary"
            data-testid="nav-icon"
          >
            <svg
              height="16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 8c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4Zm0 2c-2.67 0-8 1.34-8 4v1c0 .55.45 1 1 1h14c.55 0 1-.45 1-1v-1c0-2.66-5.33-4-8-4Z"
              />
            </svg>
          </i>
          <p
            class="text-primary dark:text-white text-medium leading-xl font-medium text-secondary"
          >
            Profile
          </p>
        </div>
      </a>
      <a
        class="flex justify-between"
        href="/sign-in"
      >
        <div
          class="flex w-full items-center gap-3 pl-8"
        >
          <i
            class="w-6 h-6 flex justify-center items-center dark:fill-white fill-secondary"
            data-testid="nav-icon"
          >
            <svg
              height="21"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14 7h-1V5c0-2.76-2.24-5-5-5S3 2.24 3 5v2H2C.9 7 0 7.9 0 9v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V9c0-1.1-.9-2-2-2Zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2ZM5 7V5c0-1.66 1.34-3 3-3s3 1.34 3 3v2H5Z"
              />
            </svg>
          </i>
          <p
            class="text-primary dark:text-white text-medium leading-xl font-medium text-secondary"
          >
            Sign In
          </p>
        </div>
      </a>
    </div>
  </nav>
</div>
`;
